core/src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -20,21 +25,50 @@ trait CassandraRecovery extends ActorLogging {
     fromSequenceNr: Long,
     toSequenceNr:   Long,
     max:            Long
-  )(replayCallback: (PersistentRepr) => Unit): Future[Unit] =
-    queries
-      .eventsByPersistenceId(
-        persistenceId,
-        fromSequenceNr,
-        toSequenceNr,
-        max,
-        replayMaxResultSize,
-        None,
-        "asyncReplayMessages",
-        someReadConsistency,
-        someReadRetryPolicy
-      )
+  )(replayCallback: (PersistentRepr) => Unit): Future[Unit] = {
+
+    // No other writes for this pid should be taking place during recovery
+    // Expecting the result set to be small (< 10 rows)so call to all() should be safe
```

Comment:
small because there is only one row per tag (per persistenceId)?
perhaps mention that in comment

Reply:
Yep :+1: unless the user tags the same persistence id with huge numbers of tags now we allow them :)